factory/README.md

```diff
@@ -0,0 +1,172 @@
+
+docker run test npm run test -b chrome
+```
+
+### In docker-compose.yml
+
+Finally, args can be specified in the docker-compose.yml file.
+
+docker-compose.yml
+
+```yml
+version: '3'
+
+services:
+  test:
+    args:
+      CHROME_VERSION: '107.0.5304.121-1'
+      EDGE_VERSION: '100.0.1185.29-1'
+      FIREFOX_VERSION: '107.0'
+    build:
+      context: .
+    command: npm run test
+```
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+docker commands
+
+```bash
+docker-compose build test
+
+docker-compose run test
+```
+
+## API
+
+The following args can be set to define what versions the cypress factory docker includes in it's final build.
+
+If no args are defined, only the default version of node will be installed. This can still be a useful container though since we will also install any required cypress apt dependencies.
+
+### NODE_VERSION
+
+The version of Node to install in the docker container. If not specified, the default version of Node (defined [here](./docker-compose.yml)) is installed. Node is required.
+
+Example: `NODE_VERSION='16.18.1'`
+
+[Node Versions](https://nodejs.org/en/download/releases)
+
+### YARN_VERSION
+
+The version of yarn to install (via npm). If not specified, Yarn is not installed.
+
+Example: `YARN_VERSION='1.22.19'`
+
+[Yarn versions](https://www.npmjs.com/package/yarn)
+
+### CYPRESS_VERSION
+
+The version of Cypress to install (via npm). If not specified, Cypress is not installed.
+
+Example: `CYPRESS_VERSION='12.1.0'`
+
+[Cypress versions](https://www.npmjs.com/package/cypress)
+
+### CHROME_VERSION
+
+The version of Chrome to install (via npm). If not specified, Chrome is not installed.
+
+Example: `CHROME_VERSION='107.0.5304.121-1'`
+
+[Chrome versions](https://www.ubuntuupdates.org/package/google_chrome/stable/main/base/google-chrome-stable)
+
+### FIREFOX_VERSION
+
+The version of Firefox to install (via npm). If not specified, Firefox is not installed.
+
+Example: `FIREFOX_VERSION='107.0'`
+
+[Firefox versions](https://download-installer.cdn.mozilla.net/pub/firefox/releases/)
+
+### EDGE_VERSION
+
+The version of Edge to install (via npm). If not specified, Edge is not installed.
+
+Example: `EDGE_VERSION='100.0.1185.29-1'`
+
+[Edge versions](https://packages.microsoft.com/repos/edge/pool/main/m/microsoft-edge-stable/)
+
+## Version Testing
+
+Due to the large amount of possible version combinations, we're not able to exhaustively test each combination of versions, nor do we block versions that are incompatible. For example, Cypress 12 removed support for node versions less than 16.16.0. You are still able to generate a container with node 16.0.0 and Cypress 12, but Cypress will fail to run. This is because the factory supports earlier versions of Cypress and must support earlier versions of node.
```

Comment:
```suggestion
Due to the large amount of possible version combinations, we're not able to exhaustively test each combination of versions, nor do we block versions that are incompatible. For example, Cypress 12 removed support for node versions less than 16.0.0. You are still able to generate a container with node 14.0.0 and Cypress 12, but Cypress will fail to run. This is because the factory supports earlier versions of Cypress and must support earlier versions of node.
```